test/location.coffee

```diff
@@ -613,3 +613,14 @@ test "Verify all tokens get a location", ->
     tokens = CoffeeScript.tokens testScript
     for token in tokens
         ok !!token[2]
+
+test 'Values with properties end up with a location that includes the properties', ->
+  source = '''
+    a.b
```

Comment:
In addition to this test, perhaps another that has a bit more code around it? Perhaps with `new`? So that we have more confidence that the location data is correct.